storage/statedb/database.go

```diff
@@ -308,20 +308,24 @@ func (t trieNodeHasher) Sum64(key string) uint64 {
 // NewDatabase creates a new trie database to store ephemeral trie content before
 // its written out to disk or garbage collected.
 func NewDatabase(diskDB database.DBManager) *Database {
-	return NewDatabaseWithCache(diskDB, 0, 0)
+	return NewDatabaseWithCache(diskDB, -1, 0)
 }
 
 // NewDatabaseWithCache creates a new trie database to store ephemeral trie content
 // before its written out to disk or garbage collected. It also acts as a read cache
 // for nodes loaded from disk.
 func NewDatabaseWithCache(diskDB database.DBManager, cacheSize int, daBlockNum uint64) *Database {
 	var trieNodeCache *bigcache.BigCache
-	if cacheSize > 0 {
+	if cacheSize >= 0 {
```

Comment:
How about adding a test of `NewDatabaseWithCache` to check if the database is generated with the cache size we expect it to be?

Reply:
I added test code for a case when cacheSize is given.
Do you have any idea to test cache size when it is scaled to physical memory?